awsx/package.json

```diff
@@ -35,7 +35,7 @@
     "@types/aws-sdk": "^2.7.0",
     "@types/jest": "^27.4.1",
     "@types/mime": "^2.0.0",
-    "@types/node": "^17.0.21",
+    "@types/node": "^18",
```

Comment:
When building I was running to a tsc compiler error, which was the same error we recently hit in `pulumi/pulumi`...

A recent change to the TypeScript compiler's dom library introduces a conflict with `@types/node` version 17. The tsc compiler errors with:

```
node_modules/@types/node/globals.d.ts:72:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'AbortSignal' must be of type '{ new (): AbortSignal; prototype: AbortSignal; abort(reason?: any): AbortSignal; timeout(milliseconds: number): AbortSignal; }', but here has type '{ new (): AbortSignal; prototype: AbortSignal; }'.

72 declare var AbortSignal: {
               ~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2071:13
    2071 declare var AbortSignal: {
                     ~~~~~~~~~~~
    'AbortSignal' was also declared here.

Found 1 error in node_modules/@types/node/globals.d.ts:72
```

Since Node.js 17 is end-of-life (and is an odd number not promoted to LTS), move to version 18 of the node types, which does not have a conflict.